docs/src/contributing_apub_api_outline.md

```diff
@@ -1,380 +1,592 @@
+| `endpoints` | no | Contains URL of shared inbox |
+| `published` | no | Datetime when the community was first created |
+| `updated` | no | Datetime when the community was last changed |
+| `publicKey` | yes | The public key used to verify signatures from this actor |
+   
+### User
+
+A person, interacts primarily with the community where it sends and receives posts/comments. Can also create and moderate communities, and send private messages to other users.
+
+Sends activities to Community: `Follow`, `Undo/Follow`, `Create`, `Update`, `Like`, `Dislike`, `Remove` (only admin/mod), `Delete` (only creator), `Undo` (only for own actions)
+
+Receives activities from Community: `Accept/Follow`, `Announce`
+
+Sends and receives activities from/to other users: `Create/Note`, `Update/Note`, `Delete/Note`, `Undo/Delete/Note` (all those related to private messages)
+
+```json
 {
-  "@context": "https://www.w3.org/ns/activitystreams",
-  "type": "Group",
-  "id": "https://instance_url/api/v1/community/today_i_learned",
-  "name": "today_i_learned"
-  "attributedTo": [ // The moderators
-    "http://joe.example.org",
-  ],
-  "followers": "https://instance_url/api/v1/community/today_i_learned/followers",
-  "published": "2014-12-31T23:00:00-08:00",
-  "summary"?: "The group's tagline",
-  "attachment: [{}] // TBD, these would be where strong types for custom styles, and images would work.
+    "@context": "https://www.w3.org/ns/activitystreams",
+    "id": "https://enterprise.lemmy.ml/u/picard",
+    "type": "Person",
+    "name": "riker",
+    "preferredUsername": "Jean-Luc Picard",
+    "summary": "The user bio",
+    "icon": {
+        "type": "Image",
+        "url": "https://enterprise.lemmy.ml/pictrs/image/DS3q0colRA.jpg"
+    },
+    "image": {
+        "type": "Image",
+        "url": "https://enterprise.lemmy.ml/pictrs/image/XenaYI5hTn.png"
+    },
+    "inbox": "https://enterprise.lemmy.ml/u/picard/inbox",
+    "endpoints": {
+        "sharedInbox": "https://enterprise.lemmy.ml/inbox"
+    },
+    "published": "2020-10-06T17:27:43.234391+00:00",
+    "updated": "2020-10-08T11:27:17.905625+00:00",
+    "publicKey": {
+        "id": "https://enterprise.lemmy.ml/u/picard#main-key",
+        "owner": "https://enterprise.lemmy.ml/u/picard",
+        "publicKeyPem": "-----BEGIN PUBLIC KEY-----\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAyH9iH83+idw/T4QpuRSY\n5YgQ/T5pJCNxvQWb6qcCu3gEVigfbreqZKJpOih4YT36wu4GjPfoIkbWJXcfcEzq\nMEQoYbPStuwnklpN2zj3lRIPfGLht9CAlENLWikTUoW5kZLyU6UQtOGdT2b1hDuK\nsUEn67In6qYx6pal8fUbO6X3O2BKzGeofnXgHCu7QNIuH4RPzkWsLhvwqEJYP0zG\nodao2j+qmhKFsI4oNOUCGkdJejO7q+9gdoNxAtNNKilIOwUFBYXeZJb+XGlzo0X+\n70jdJ/xQCPlPlItU4pD/0FwPLtuReoOpMzLi20oDsPXJBvn+/NJaxqDINuywcN5p\n4wIDAQAB\n-----END PUBLIC KEY-----\n"
+    }
 }
 ```
 
+| Field Name | Mandatory | Description |
+|---|---|---|
+| `name` | yes | Name of the actor |
+| `preferredUsername` | yes | Displayname |
```

Comment:
This is an optional field. You have it right for community, where it's not-optional, but for users its optional.

Reply:
Thanks, fixed.